src/main/java/book/service/BookApp.java

```diff
@@ -0,0 +1,216 @@
+                                request.put(owners.get(isbn).getOwnerEmail(),"Hi, I am " + user.getFirstName() + " " + user.getLastName() + " I need book from you and book isbn number is " +isbn + " please contact me on my number    " + user.getMobileNumber());
+                                changeBookStatus(bookName,isbn,bookStore,BookStatus.TAKEN);
+                                bookShare.put(isbn,new OwnerDTO(user.getFirstName(),user.getEmail()));
+                            }
+                        }
+                    }else{
+                        System.out.println("book not present in store to request or taken away");
+                    }
+                } else if(choice == 6) {
+                     System.out.println("Enter the name of book you want to return");
+                     String select = scanner.next();
+                     if(bookStore.containsKey(select)){
+                     System.out.println(" please enter the isbn number of your book");
+                     int isbn = scanner.nextInt();
+                     if (bookShare.containsKey(isbn)) {
+                         System.out.println("Hello   " + bookShare.get(isbn).getOwnerName() + " Thanks for visiting we will mark book status returned");
+                         changeBookStatus(select,isbn,bookStore,BookStatus.AVAILABLE);
+                         bookShare.remove(isbn);
+                         request.put(owners.get(isbn).getOwnerEmail(),"Hi    " + owners.get(isbn).getOwnerName() + " ,   " + user.getFirstName() + " " + "Returned book with isbn  " + isbn);
+                     }
+                     }else{
+                         System.out.println(" no such books to return");
+                     }
+                } else if(choice == 7) {
+                    logout = true;
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+
+    
+    public Book addBook(String name) throws ParseException {
+        Scanner scanner = new Scanner(System.in);
+        Random random = new Random();
+        int isbn = random.nextInt();
+        System.out.println("Enter the number of authors you want to add");
+        int number = scanner.nextInt();
+        List<String > authors = new ArrayList<>();
+        for(int index = 1; index<= number ; index++){
+            authors.add(scanner.next());
+        }
+        Date date = convertDate(java.time.LocalDate.now().toString());
+        System.out.println("Enter the two keywords");
+        List<String>  key = new ArrayList<>();
+        key.add(scanner.next());
+        key.add(scanner.next());
+        return  new Book(isbn,name,authors,date,key, BookStatus.AVAILABLE.getName());
+    }
+
+
+
+
+    public List<Book> byAuthor(Map<String,List<Book>> bookStore,String authorName){
+        List<List<Book>> sortAuthor = bookStore.values().stream().collect(Collectors.toList());
```

Comment:
Use Java-8 stream APIs to filter and return the books, throw user defined exceptions if match fails